src/php_spx.c

```diff
@@ -557,12 +641,37 @@ static void profiling_handler_ex_hook_before(void)
 
 static void profiling_handler_ex_hook_after(void)
 {
+    if (!context.profiling_handler.profiler) {
+        return;
+    }
+
+    if (context.profiling_handler.leaving_start) {
+        context.profiling_handler.leaving_start = 0;
+
+        if (context.config.builtins) {
+            /*
+                Nothing to instrument: we are leaving the spx_profiler_start() function
+            */
+            return;
+        }
+    }
+
 #ifdef USE_SIGNAL
     context.profiling_handler.sig_handling.probing = 1;
 #endif
 
     context.profiling_handler.profiler->call_end(context.profiling_handler.profiler);
 
+    context.profiling_handler.depth--;
+
+    if (!context.config.auto_start && context.profiling_handler.depth == 0) {
+        /*
+            We are leaving the function where the profiling was initially started
+            so we implicitly stop it.
```

Comment:
:-1: Should only implicitly stop when the process exits, for the reason I gave as a comment in the main thread.

Reply:
Yes I got your point. The current behavior is also, tbh, the easiest path to achieve runtime starts / stops. But you're right, especially with your example above which highlights the fact that it even makes impossible to hide spx_profiler_start behind a more abstract function.